File: benchmarks/oopsla25/subjects/341.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f341(const char* s) {
    int i = 0;
    while (i < strlen(s)) {
        char a = s[i];
        char x = s[i+1];
        char b = s[i+2];
        assert(a == 'a');
        assert(b == 'b');
        i++;
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f341(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/341.py
import sys

def f341(s: str):
  i = 0
  while i < len(s):
    a,x,b = s[i:i+3]
    assert a == "a"
    assert b == "b"
    i += 1

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f341(my_string)
